### PR TITLE
fix: also check exceptions/stack trace when filtering log messages

### PR DESCRIPTION
**Pull request by Masterchen09**
Unfortunately the log filter from my PR [here](https://github.com/datahub-project/datahub/pull/10043) does not work because the string "Config does not exist: file:///etc/datahub/plugins/frontend/auth/user.props" is not part of the (formatted) log message, instead this is part of the exceptions/stack trace.

As a solution, we could use the string "Exception starting propertyUserStore /etc/datahub/plugins/frontend/auth/user.props" for the filter, but then all exceptions related to the user.props file would be excluded, therefore I have enhanced the LogMessageFilter to also check the exceptions resp. the stack trace, then the filter from my previous PR is working (and this time I have verified that it really works 😇).

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved log message filtering to handle throwable proxies, ensuring more accurate and reliable log management.

### CODE REVIEW

File: metadata-utils/src/main/java/com/linkedin/metadata/utils/log/LogMessageFilter.java [hunk 1]
```diff
@@ -21,7 +23,21 @@ public FilterReply decide(ILoggingEvent event) {
       return FilterReply.NEUTRAL;
     }
 
-    if (this.excluded.stream().anyMatch(message -> event.getFormattedMessage().contains(message))) {
+    final String formattedMessage = event.getFormattedMessage();
+
+    IThrowableProxy throwbleProxy = event.getThrowableProxy();
```
Comment (jjoyce0510) [inline metadata-utils/src/main/java/com/linkedin/metadata/utils/log/LogMessageFilter.java:28]: typo: throwableProxy

Comment (jjoyce0510) [reply metadata-utils/src/main/java/com/linkedin/metadata/utils/log/LogMessageFilter.java:28]: ill edit this for u then we can merge

**Review by jjoyce0510 — APPROVED** [review]
LGTM once ci + linting passes

**Comment by Masterchen09** [comment]
Assigning null directly to throwableString before the if caused throwableString to no longer be effectively final (although null is definitely better than an empty string) -> My last commit should fix this.

**Comment by jjoyce0510** [comment]
Aiming to merge these PRs today once I get CI passing.

**Pull request merged**